lvsfunc/util.py

```diff
@@ -102,6 +103,45 @@ def get_prop(frame: vs.VideoFrame, key: str, t: Type[T]) -> T:
     return prop
 
 
+def select_frames(clips: Union[vs.VideoNode, List[vs.VideoNode]],
+                  indicies: Union[List[int], List[Tuple[int, int]]]) -> vs.VideoNode:
+    """
+    Select frames from one or more clips at specified indices.
+
+    :param clips:   A clip or a list of clips to select the frames from
+    :param indexes: Indices of frames to select.
+
+                    Provide a list of indices for a single clip, or for multiple clips, a list of tuples in the
+                    form ``(clip_index, frame_index)``
+
+    :return:        The selected frames in a single clip
+    """
+    clips = [clips] if isinstance(clips, vs.VideoNode) else clips
+    for pos, index in enumerate(indicies):
```

Comment:
removing the union removes the issues with this code, otherwise
```python
for index in indices:
    if isinstance(index, int):
        clip_idx, frame_idx = (0, index)
    else:
        clip_idx, frame_idx = index
```
then
```python
if isinstance(index, int):
    clip_idx, frame_idx = (0, indices[n])
else:
    clip_idx, frame_idx = indices[n]
return clips[clip_idx][frame_idx]
```
in the frameeval

Reply:
I would rather not have any more logic than is strictly necessary within the frame eval. However I agree that expanding this into multiple lines would help with both readability and mypy.

Reply:
that makes sense for VS filters, but doing a typecheck and conditional tuple construction is virtually free in the context of video processing